src/include/homestore/replication/repl_dev.h

```diff
@@ -399,6 +412,39 @@ class ReplDev {
     ReplDev() = default;
     virtual ~ReplDev() { detach_listener(); }
 
+    /// @brief Allocates blkids from the storage engine to write the value into. Storage
+    /// engine returns a blkid_list in cases where single contiguous blocks are not
+    /// available.
+    ///
+    /// @param data_size - Size of the data.
+    /// @param hints - Specify block allocation hints.
+    /// @param out_blkids - List of bilkid's which may not be contiguous.
+    virtual std::error_code alloc_blks(uint32_t data_size, const blk_alloc_hints& hints,
+                                       std::vector< MultiBlkId >& out_blkids) = 0;
```

Comment:
looks like we are already exposing `std::vector< MultiBlkId >& out_blkids`, not `std::vector< BlkId >& out_blkids`, right?